src/containers/CreatePipelineRun/CreatePipelineRun.js

```diff
@@ -0,0 +1,527 @@
+      imageresourcename: this.state.imageName.value,
+      pipelineruntype: this.state.helmPipeline.value ? 'helm' : '',
+      helmsecret: this.state.helmSecret.value
+    };
+    const promise = this.props.createPipelineRun({
+      payload,
+      namespace: this.getNamespace()
+    });
+    promise.then(headers => {
+      if (headers) {
+        const url = headers.get('Content-Location');
+        const pipelineRunName = url.substring(url.lastIndexOf('/') + 1);
+        const finalURL = `/pipelines/${this.getPipeline()}/runs/${pipelineRunName}`;
+        this.reset();
+        this.props.onSuccess({ name: pipelineRunName, url: finalURL });
+      }
+    });
+  }
+
+  onClose() {
+    this.reset();
+    this.props.onClose();
+  }
+
+  getPipeline() {
+    return this.props.pipelineName || this.state.pipeline.value;
+  }
+
+  getNamespace() {
+    if (this.props.namespace) {
+      return this.props.namespace;
+    }
+    if (this.state.namespace.value !== '') {
+      return this.state.namespace.value;
+    }
+    if (this.props.selectedNamespace !== ALL_NAMESPACES) {
+      return this.props.selectedNamespace;
+    }
+    return '';
+  }
+
+  checkValidation(key, value) {
+    const { required, regexp } = formValidation[key];
+    const invalid =
+      (required && value === '') ||
+      (!!regexp && !regexp.test(value) && !(!required && value === ''));
+    this.setState(state => {
+      return {
+        [key]: {
+          ...state[key],
+          invalid
+        }
+      };
+    });
+    return !invalid;
+  }
+
+  checkFormValidation() {
+    const reducer = (acc, key) => {
+      if (key === 'namespace') {
+        return this.checkValidation(key, this.getNamespace()) && acc;
+      }
+      if (key === 'pipeline') {
+        return this.checkValidation(key, this.getPipeline()) && acc;
+      }
+      return this.checkValidation(key, this.state[key].value) && acc;
+    };
+    return Object.keys(formValidation).reduce(reducer, true);
+  }
+
+  reset() {
+    this.setState(() => {
+      return initialState;
+    });
+  }
+
+  render() {
+    const { pipelineName, errorMessage, open, namespace } = this.props;
+
+    return (
+      <>
+        <Form onSubmit={this.onSubmit}>
+          <ComposedModal
+            className="create-pipelinerun"
+            onClose={this.onClose}
+            open={open}
+          >
+            <ModalHeader
+              id="create-pipelinerun--header"
+              title="Create PipelineRun"
+              label={pipelineName}
+              closeModal={this.onClose}
+            />
+            <ModalBody>
+              {(() => {
+                if (errorMessage) {
+                  return (
+                    <InlineNotification
+                      kind="error"
+                      title="Error creating PipelineRun"
+                      subtitle={errorMessage}
+                    />
+                  );
+                }
+                return null;
```

Comment:
ESLint complains if I don't have it :/
`Expected to return a value at the end of arrow function.eslint(consistent-return)`

Reply:
instead of this:

```js
{(() => {
  if (this.state.createdPipelineRun) {
    return (
      <InlineNotification
        kind="success"
        title="Successfully created a PipelineRun"
        subtitle={
          <Link to={this.state.createdPipelineRunURL}>
            {this.state.createdPipelineRunName}
          </Link>
        }
        hideCloseButton={false}
      />
    );
  }
  return null;
})()}
```

you should be able to do:

```js
{this.state.createdPipelineRun && (
  <InlineNotification
    kind="success"
    title="Successfully created a PipelineRun"
    subtitle={
      <Link to={this.state.createdPipelineRunURL}>
        {this.state.createdPipelineRunName}
      </Link>
    }
    hideCloseButton={false}
  />
)}
```

Reply:
Awesome, thank you! 😄